backend/src/routes/events.ts

```diff
@@ -39,14 +39,65 @@ type ServerSideEventData = WatchEvent | SettingsEvent | { type: 'START' | 'LOADE
 
 let requests: { cancel: () => void }[] = []
 
-const resourceCache: {
+export function getKubeResources(kind: string, apiVersion: string) {
+  const option = { apiVersion, kind }
+  const apiVersionPlural = apiVersionPluralFn(option)
+  return Object.values(resourceCache[apiVersionPlural] || {}).map((event) => {
+    return event.resource
+  })
+}
+
+export async function getAuthorizedLocalResources(resources: IResource[], token: string): Promise<IResource[]> {
+  const authorized: IResource[] = []
+  const queue = resources.map((resource) => {
+    return canListResources(token, resource)
+      .then((allowResource) => (allowResource ? resource : undefined))
+      .catch((err: unknown) => undefined) as Promise<IResource>
+  })
+  while (queue.length) {
+    const resource = await queue.shift()
+    if (resource) {
+      authorized.push(resource)
+    }
+  }
+  return authorized
+}
+
+export async function getAuthorizedRemoteResources(resources: IResource[], token: string): Promise<IResource[]> {
+  if (
+    await canAccess(
+      { kind: 'ManagedClusterView', apiVersion: 'view.open-cluster-management.io/v1beta1' },
+      'create',
+      token
+    )
+  ) {
+    return resources
+  }
+  return []
+}
```

Comment:
This needs to check the specific namespace for the cluster that the resources come from.